mypy/build.py

```diff
@@ -580,33 +581,43 @@ def plugin_error(message: str) -> None:
     errors.set_file(options.config_file, None)
     for plugin_path in options.plugins:
         # Plugin paths are relative to the config file location.
-        plugin_path = os.path.join(os.path.dirname(options.config_file), plugin_path)
-
-        if not os.path.isfile(plugin_path):
-            plugin_error("Can't find plugin '{}'".format(plugin_path))
-        plugin_dir = os.path.dirname(plugin_path)
-        fnam = os.path.basename(plugin_path)
-        if not fnam.endswith('.py'):
+        func_name = 'plugin'
+        plugin_dir = None  # type: Optional[str]
+        if ':' in os.path.basename(plugin_path):
+            plugin_path, func_name = plugin_path.rsplit(':', 1)
+        if plugin_path.endswith('.py'):
+            plugin_path = os.path.join(os.path.dirname(options.config_file), plugin_path)
+            if not os.path.isfile(plugin_path):
+                plugin_error("Can't find plugin '{}'".format(plugin_path))
+            plugin_dir = os.path.dirname(plugin_path)
+            fnam = os.path.basename(plugin_path)
+            module_name = fnam[:-3]
+            sys.path.insert(0, plugin_dir)
+        elif re.search(r'\/', plugin_path):
```

Comment:
Thanks for catching that.  What I intended to do was this:

```python
re.search(r'[\\/]', plugin_path) 
```

I considered `os.sep`, but decided against it because it's bound to the current OS.  I think it's more correct to detect *anything* that looks like a file path, but I'll let you make the call.

Reply:
That is a good point. I think your proposed solution is good.